packit/config.py

```diff
@@ -287,6 +289,78 @@ def get_deprecated_key(raw_dict: dict, new_key_name: str, old_key_name: str):
     def validate_dict(cls, raw_dict: dict) -> None:
         jsonschema.validate(raw_dict, PACKAGE_CONFIG_SCHEMA)
 
+    def run_action(
```

Comment:
I've played with other possibilities as well, but:

- It needs `package_config.actions`.
- We need to use it from `PackitAPI` and `Upstream`/`Distgit`.

It is quite difficult to avoid unnecessary data passing through the whole code.

The common ancestor of the `Upstream`/`Distgit` may be a better place. I'll try to play with it once more.

Reply:
...or local project maybe and accept actions as an argument?